buildenv/jenkins/common/pipeline-functions

```diff
@@ -555,4 +586,99 @@ def generate_test_jobs(TARGET_NAMES, SPEC, ARTIFACTORY_SERVER, ARTIFACTORY_REPO)
+    }
+
+    // Setup TESTS_TARGETS
+    switch ("${PARSED_COMMENT[1]}") {
+        case "compile":
+            if (PARSED_COMMENT.size() < 4) {
+                error("Pull Request trigger comment needs to specify PLATFORM and SDK_VERSION")
+            }
+            TESTS_TARGETS = 'none'
+            OFFSET = 0
+            break
+        case "test":
+            if (PARSED_COMMENT.size() < 5) {
+                error("Pull Request trigger comment needs to specify TESTS_TARGET, PLATFORM and SDK_VERSION")
+            }
+            TESTS_TARGETS = "${PARSED_COMMENT[2]}"
+            if ("${TESTS_TARGETS}" == "all") {
+                error("Test Target 'all' not allowed for Pull Request builds")
+            }
+            OFFSET = 1
+            break
+        default:
+            error("Unable to parse trigger comment. Unknown build option:'${PARSED_COMMENT[1]}'")
+    }
+    echo "TESTS_TARGETS:'${TESTS_TARGETS}'"
+
+    // Setup JDK VERSIONS
+    switch (ghprbGhRepository) {
+        case   ~/.*openj9-openjdk-jdk.*/:
+            TMP_VERSION = ghprbGhRepository.substring(ghprbGhRepository.indexOf('-jdk')+4)
+            if ("${TMP_VERSION}" == "") {
+                TMP_VERSION = 'next'
+            }
+            RELEASES.add(TMP_VERSION)
+            MIN_COMMENT_SIZE = 3
+            break
+        case "eclipse/openj9":
+            TMP_VERSIONS = PARSED_COMMENT[3+OFFSET].tokenize(',')
+            TMP_VERSIONS.each { VERSION ->
+                echo "VERSION:'${VERSION}'"
+                if ("${VERSION}" == "all") {
+                    error("JDK version 'all' not allowed for Pull Request builds\nExpected 'jdkX' where X is one of ${CURRENT_RELEASES}")
+                }
+                if (!("${VERSION}" ==~ /jdk.*/)) {
+                    error("Incorrect syntax for requested JDK version:'${VERSION}'\nExpected 'jdkX' where X is one of ${CURRENT_RELEASES}")
+                }
+                if (CURRENT_RELEASES.contains(VERSION.substring(3))) {
+                    RELEASES.add(VERSION.substring(3))
+                } else {
+                    error("Unsupported JDK version or incorrect syntax:'${VERSION}'\nExpected 'jdkX' where X is one of ${CURRENT_RELEASES}")
+                }
+            }
+            MIN_COMMENT_SIZE = 4
+            break
+        default:
+            error("Unknown source repo:'${ghprbGhRepository}'")
+    }
+
+    echo "RELEASES:'${RELEASES}'"
+
+    if (PARSED_COMMENT.size() < MIN_COMMENT_SIZE) { error("Pull Request trigger comment does not contain enough elements.")}
+
+    // Setup PLATFORMS
+    PARSED_PLATFORMS = PARSED_COMMENT[2+OFFSET].tokenize(',')
+    PARSED_PLATFORMS.each { SHORT ->
+        LONG_PLATFORM = SHORT_NAMES["${SHORT}"]
+        if (LONG_PLATFORM) {
+            PLATFORMS.add(LONG_PLATFORM)
+        } else {
+            error("Unknown PLATFORM short:'${SHORT}'\nExpected one of:${SHORT_NAMES}")
+        }
+    }
+    echo "PLATFORMS:'${PLATFORMS}'"
```

Comment:
How is 'depends' handled?

Reply:
Added a comment to comment block at the beginning of `setup_pull_request()`